zerver/migrations/0424_backfill_remote_realmauditlog_extradata_to_json_field.py

```diff
@@ -0,0 +1,72 @@
+# Generated by Django 4.0.7 on 2022-09-30 20:30
+
+import ast
+from typing import Type
+
+from django.db import migrations
+from django.db.backends.postgresql.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from django.db.models import JSONField, Model
+from django.db.models.functions import Cast, JSONObject
+
+USER_FULL_NAME_CHANGED = 124
+BATCH_SIZE = 5000
+
+
+def do_bulk_backfill_extra_data(
+    audit_log_model: Type[Model], id_lower_bound: int, id_upper_bound: int
+) -> None:
+    # First handle the special case for audit logs with the USER_FULL_NAME_CHANGED event, which stores
+    # the full name not as str(dict()) but a plain str.
+    audit_log_model.objects.filter(
+        event_type=USER_FULL_NAME_CHANGED, id__range=(id_lower_bound, id_upper_bound)
+    ).update(extra_data_json=JSONObject(old_name="extra_data"))
```

Comment:
I guess we also need to decide if we're fixing this up to be `RealmAuditLog.OLD_VALUE="extra_data"` (modulo not being able to directly use that constant, because we're in a migration).  The reason not to is that we don't have a NEW_VALUE to go along with it -- but being slightly more standard seems useful, and we can switch the USER_FULL_NAME_CHANGED codepath to, going forward, give both OLD_VALUE and NEW_VALUE.

Reply:
OK, I set that up with `RealmAuditLog.OLD_VALUE`. I think we can later discuss if we are switching to the dedicated columns for this purpose.

Reply:
My point wasn't about dedicated columns, but about if we should do:
```python3
def do_change_full_name(
    user_profile: UserProfile, full_name: str, acting_user: Optional[UserProfile]
) -> None:
    old_name = user_profile.full_name
    user_profile.full_name = full_name
    user_profile.save(update_fields=["full_name"])
    event_time = timezone_now()
    RealmAuditLog.objects.create(
        realm=user_profile.realm,
        acting_user=acting_user,
        modified_user=user_profile,
        event_type=RealmAuditLog.USER_FULL_NAME_CHANGED,
        event_time=event_time,
        extra_data=old_name,
        extra_data={RealmAuditLog.OLD_VALUE: old_name, RealmAuditLog.NEW_VALUE: full_name},
    )
```
...where we start setting NEW_VALUE as well.